packages/roosterjs-content-model-plugins/lib/autoFormat/AutoFormatPlugin.ts

```diff
@@ -153,19 +165,18 @@ export class AutoFormatPlugin implements EditorPlugin {
                                 shouldHyphen = transformHyphen(previousSegment, paragraph, context);
                             }
 
-                            return (
-                                keyboardListTrigger(
-                                    model,
-                                    paragraph,
-                                    context,
-                                    autoBullet,
-                                    autoNumbering
-                                ) ||
-                                shouldHyphen ||
-                                shouldLink
-                            );
+                            apiName = getApiName(shouldList, shouldHyphen);
+
+                            return shouldList || shouldHyphen || shouldLink;
                         }
                     );
+                    if (apiName) {
+                        editor.triggerEvent('contentChanged', {
```

Comment:
We should let `formatTextSegmentBeforeSelectionMarker` trigger the event for us, we just need to pass in the change source. You can pass in from the 3rd parameter 
![image](https://github.com/microsoft/roosterjs/assets/23065085/764a98f8-e091-4c61-a6b0-dfc6c0655a34)

Reply:
Since the apiName can only be retrieved after callback, we can use a trick here:

```ts
const option: FormatContentModelOptions = {
  source: 'autoFormat' // BTW, Can we add this into `ChangeSource` in `roosterjs-content-model-dom`?
}

formatTextSegmentBeforeSelectionMarker (editor, model=> {
  ...
  option.formatApiName = apiName;
}, option);
```